### DISCUSSION TITLE
Is there a repository of Community-Made SLEIGH Instruction Sets anywhere?

### DISCUSSION THREAD

**Discussion by f-raZ0R** [NationalSecurityAgency/ghidra#7247] [Q&A] [answered]
So, I am decompiling a 3DS game, which uses armv6, as well as the VFP Coprocessor Unit for floating point math instructions. I am... completely missing all of the coprocessor instructions, as well as the 3ds's kernel syscall functions/datatypes. After doing a _lot_ of digging, I still can't... find any plugin or module anything else that I could use to import these instructions into my project?
I would rather not reinvent the wheel here and try to define all these instructions myself, so... where do I even find these?
The VFP floating point math instructions are a much bigger concern to be lacking, and are a much higher priority to have, for me.

**Comment by GhidorahRex** [comment]
Which ARM variant are you using? We have VFP instructions in the ARM language. Do you have some sample instructions that are not disassembling?

**Reply by f-raZ0R** [reply]
ARM:LE:32:v6

having some trouble finding resources on what instructions these actually map to (it is easy to find out what they do, but not easy to find any documentation on what bytes they correspond to), but here are a couple screenshots
![Screenshot (1815)](https://github.com/user-attachments/assets/e6a334fe-5180-4329-85d1-021988ef5d0b)
![Screenshot (1817)](https://github.com/user-attachments/assets/f9e3269d-14d8-47e8-b13f-4ecebe524823)
![Screenshot (1818)](https://github.com/user-attachments/assets/261554cf-fcb0-45e2-aa70-8308c51ab1ff)

**Reply by f-raZ0R** [reply]
@GhidorahRex

**Reply by GhidorahRex** [reply, marked as answer]
@f-raZ0R So they're disassembling, but only as the cdp instructions, which is not correct. That last one, for example, should be `vsqrt.f32 s0, s1` The other ones in your screenshot should be `vmul.f32`.  It looks like the issue is that these two VFP instructions are being incorrectly wrapped by `@if defined(SIMD)` instead of `@if defined(VFPv2) || defined(VFPv3) || defined(SIMD)`. I went through the other VFPv2 instructions that I could find and didn't see any others that were amiss.

I've attached a fixed version of the `ARMneon.sinc` file. Can you check and see if that fixes all of the VFP instructions in your binary?
[ARMneon.sinc.txt](https://github.com/user-attachments/files/18040977/ARMneon.sinc.txt)

**Reply by f-raZ0R** [reply]
It fixes the cases of vmul.f32, but for the case of the vsqrt in the math_sqrt method, it is instead being interpreted as vmul.f32 s0, s2, s0?

**Reply by GhidorahRex** [reply]
When I use the file above, with the bytes `e0 0a b1 ee` I get `vsqrt.f32  s0,s1`. `vmul.f32 s0, s2, s0` would be the bytes `00 0a 21 ee`. If I comment out the `vsqrt.f32` constructor it reverts to cdp. I'm not sure how to get it to parse as `vmul.f32`

**Reply by f-raZ0R** [reply]
Oh interesting, pressing C to clear the bytes, and then D to reassemble them corrects the instruction to the `vsqrt` it should be.
Might be due to loading it on a pre-existing project?

**Reply by f-raZ0R** [reply]
A lot of registers used in `vmul` instructions are also incorrect, but correct themselves when re-assembled
